packages/react/src/components/tag/tag.tsx

```diff
@@ -167,87 +233,111 @@ const Container = styled.span<ContainerProps>`
         margin-left: var(--spacing-1x);
     }
 
-    ${DeleteButton} {
-        margin-right: calc(-1 * var(--spacing-half));
-    }
+    ${({ $selected, theme }) => $selected && css`
+        background-color: ${theme.component['tag-selected-background-color']};
+        border: 1px solid ${theme.component['tag-selected-border-color']};
+    `}
 
     ${getClickableStyle};
-
-    ${focus};
 `;
 
 export const Tag = forwardRef(({
     className,
     iconName,
-    onClick,
-    onDelete,
     size = 'medium',
+    color = 'default',
+    selected = false,
     value,
+    onClick,
+    onRemove,
 }: TagProps, ref: Ref<HTMLElement>) => {
-    if (onClick && onDelete) {
-        throw new Error('Only one of onClick or onDelete can be provided.');
-    }
-
     const { t } = useTranslation('tag');
     const { isMobile } = useDeviceContext();
+
+    const isRemovable = !!onRemove;
+    const isClickable = !!onClick;
+    const hasIcon = !!iconName;
+    const currentColor = (isRemovable || isClickable || hasIcon) ? 'default' : color;
+    const [isSelected, setSelected] = useState(hasDefaultColor(currentColor) && selected);
     const hasIconLabel = !(value.label.toLowerCase() === iconName?.toLowerCase());
 
+    useEffect(() => {
```

Comment:
Je pense que tu pourrais utiliser cette stratégie pour éviter d'avoir un `useEffect`:
https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes